native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+                add_param_star(a, star)
+            }
+
+        rule param_no_default() -> Param<'a>
+            = a:param() c:lit(",") { add_param_default(a, None, Some(c)) }
+            / a:param() &lit(")") {a}
+
+        rule param_with_default() -> Param<'a>
+            = a:param() def:default() c:lit(",") {
+                add_param_default(a, Some(def), Some(c))
+            }
+            / a:param() def:default() &lit(")") {
+                add_param_default(a, Some(def), None)
+            }
+
+        rule param_maybe_default() -> Param<'a>
+            = a:param() def:default()? c:lit(",") {
+                add_param_default(a, def, Some(c))
+            }
+            / a:param() def:default()? &lit(")") {
+                add_param_default(a, def, None)
+            }
+
+        rule param() -> Param<'a>
+            = n:name() a:annotation()? {
+                Param {name: n, annotation: a, ..Default::default() }
+            }
+
+        rule annotation() -> Annotation<'a>
+            = col:lit(":") e:expression() {
+                make_annotation(col, e)
+            }
+
+        rule default() -> (AssignEqual<'a>, Expression<'a>)
+            = eq:lit("=") ex:expression() {
+                (make_assign_equal(eq), ex)
+            }
+
+        rule if_stmt() -> If<'a>
+            = i:lit("if") a:named_expression() col:lit(":") b:block() elif:elif_stmt() {
+                make_if(i, a, col, b, Some(OrElse::Elif(elif)), false)
+            }
+            / i:lit("if") a:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_if(i, a, col, b, el.map(OrElse::Else), false)
+            }
+
+        rule elif_stmt() -> If<'a>
+            = i:lit("elif") a:named_expression() col:lit(":") b:block() elif:elif_stmt() {
+                make_if(i, a, col, b, Some(OrElse::Elif(elif)), true)
+            }
+            / i:lit("elif") a:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_if(i, a, col, b, el.map(OrElse::Else), true)
+            }
+
+        rule else_block() -> Else<'a>
+            = el:lit("else") col:lit(":") b:block() {
+                make_else(el, col, b)
+            }
+
+        rule while_stmt() -> While<'a>
+            = kw:lit("while") test:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_while(kw, test, col, b, el)
+            }
+
+        rule for_stmt() -> For<'a>
+            = f:lit("for") t:star_targets() i:lit("in") it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(None, f, t, i, it, c, b, el)
+            }
+            / asy:tok(Async, "ASYNC") f:lit("for") t:star_targets() i:lit("in")
+                it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(Some(asy), f, t, i, it, c, b, el)
+            }
+
+        rule with_stmt() -> With<'a>
```

Comment:
[compat] need to add parenthesized with statements, which also allow trailing commas (later)